std/concurrency.d

```diff
@@ -2397,12 +2397,12 @@ version (unittest)
 private @property Mutex initOnceLock()
 {
     __gshared Mutex lock;
```

Comment:
We need to fix `initOnce` so that it takes a `shared(Mutex)` and deprecate the overload that takes a non-shared one.
I fixed core.sync.mutex.Mutex so that it works well it shared one or two releases ago. I think `initOnce` taking a non-shared `Mutex` was simply because  `shared(Mutex)` didn't work at all at the time the function was added.